ppr-ui/src/components/results/Results.vue

```diff
@@ -0,0 +1,180 @@
+<template>
+  <div class="main-results-div">
+    <v-row v-if="searched" no-gutters class="pa-3 result-info">
+      <v-col v-if="totalResultsLength !== 0">
+        {{ totalResultsLength }} registrations found |
+        <b>{{ exactMatchesLength }} exact matches</b> |
+        <b>selected: {{ selectedLength }}</b>
+      </v-col>
+      <v-col v-else>
+        {{ totalResultsLength }} registrations found
+      </v-col>
+    </v-row>
+    <v-row v-if="totalResultsLength !== 0" no-gutters>
+      <v-container fluid no-gutters>
+        <v-row no-gutters>
+          <v-col class="auto">
+            <v-data-table v-if="results"
+                          class="results-table"
+                          height="20rem"
+                          hide-default-footer
+                          fixed
+                          fixed-header
+                          :headers="headers"
+                          :items="results"
+                          :item-class="getClass"
+                          item-key="vehicleCollateral.serialNumber"
+                          multi-sort
+                          show-select
+                          return-object
+                          v-model="selected">
+            </v-data-table>
+          </v-col>
+        </v-row>
+      </v-container>
+    </v-row>
+    <v-row v-else no-gutters justify="center" class="pa-3">
+      <div class="no-results-info white">
+        <v-row no-gutters justify="center" class="pt-10">
+          No registrations were found for the Serial Number:
+        </v-row>
+        <v-row no-gutters justify="center">
+          <b>{{ searchValue }}</b>
+        </v-row>
+      </div>
+    </v-row>
+  </div>
+</template>
+
+<script lang="ts">
+import { computed, defineComponent, reactive, toRefs, watch } from '@vue/composition-api'
+import { useGetters } from 'vuex-composition-helpers'
+
+import { tableHeaders } from '@/resources'
+import { SearchResponseIF, SearchResultIF, TableHeadersIF } from '@/interfaces' // eslint-disable-line no-unused-vars
+import { MatchTypes } from '@/enums'
+
+export default defineComponent({
```

Comment:
For awhile now we have been writing class based components opposed to using object notation. 
This approach certainly isn't wrong and i think it's more or less a preference.
I think we opted for class based components due to the way it worked well with TypeScript. 

Here is a really simple example that utilizes a lot of Vue Property Decorators, it just makes for very easy to read components.

https://github.com/bcgov/bcrs-shared-components/blob/master/src/components/DetailComment/DetailComment.vue

Not saying you should change anything, just making aware of the approach we are currently taking in our other UI's, utilizing `vue-property-decorator`.

Reply:
answered below